### DISCUSSION TITLE
Task builder state machine not visible in decompilers

### DISCUSSION THREAD

**Discussion by kerams** [dotnet/fsharp#14245] [Q&A] [answered]
Given

```fsharp
let c () = task {
    printfn "yup"
}
```

I can see the state machine struct in [Sharplab](https://sharplab.io/#v2:DYLgZgzgPsCmAuACAxogFASkQXkfAhhANaIDeAsAFCI2IAOATgJYB28YLiARAJ4CudLlQC+QA===), but dnSpy, ILSpy (and I think dotPeek too, last I checked) refuse to show it.

![image](https://user-images.githubusercontent.com/5063478/200113148-259de4da-1028-4af0-ac5f-e307ee115d5c.png)

It's clickable, but nothing happens, and I can't find the struct anywhere in the assembly explorer panel either. Any clue what's going on? Is it a problem with assembly metadata?

**Comment by kerams** [comment, marked as answer]
Turns out there's a checkbox that needs to be ticked. In dnSpyEx it's `Options->Decompiler->Show hidden compiler generated types and methods`.

**Reply by abelbraaksma** [reply]
Likewise, `dotPeek` has a setting for this, and probably `ILSpy` as well.
